2_process/src/lake_sequences_mntoha.py

```diff
@@ -0,0 +1,228 @@
+                       drivers_file,
+                       clarity_file,
+                       ice_flags_file,
+                       config):
+    """
+    Assemble features and observed temperatures from one lake into equal-length
+    sequences for training/testing.
+
+    :param site_id: ID of lake to process
+    :param metadata_augmented_file: Lake metadata file, augmented with elevation
+    :param obs_interpolated_file: Temperature observations file, interpolated
+        to LSTM depths
+    :param drivers_file: Meteorological drivers csv for this lake
+    :param clarity_file: Time-varying lake clarity csv for this lake
+    :param ice_flags_file: Time-varying ice flags csv for this lake
+    :param config: Snakemake config for 2_process
+    :returns: Numpy array of sequences with shape (# sequences,
+        sequence_length, # features + depths)
+
+    """
+    # Settings for forming LSTM training sequences
+    # Length of each sequence in days
+    sequence_length = config['sequence_length']
+    # Days between the start of one sequence and the start of the next
+    sequence_offset = config['sequence_offset']
+    # Number of days before 1st obs, e.g., 100 days
+    spinup_time = config['spinup_time'] 
+    # Discretized lake depths to use for LSTM
+    depths = np.array(config['depths'])
+
+    # All temperature observations interpolated to discretized depths
+    obs_interpolated = pd.read_csv(obs_interpolated_file, parse_dates=['date'])
+    # Observations for this lake
+    lake_obs_interpolated = obs_interpolated[obs_interpolated.site_id==site_id].copy()
+    # Only form sequences if there are any observations in this lake
+    if len(lake_obs_interpolated) == 0:
+        lake_sequences = np.empty((0))
+    else:
+        # Read metadata with elevation
+        lake_metadata_augmented = pd.read_csv(metadata_augmented_file)
+        # NOTE: If there are duplicate sites in lake_metadata_augmented,
+        # only the first will be used.
+        # If there are no matches, an IndexError will be raised.
+        lake = lake_metadata_augmented[lake_metadata_augmented.site_id==site_id].iloc[0,:]
+
+        # Create one large DataFrame of temperatures at all days and all discretized depths
+        obs_full = all_dates_depths(lake_obs_interpolated,
+                                    depths,
+                                    pad_before_days=spinup_time,
+                                    min_days=sequence_length)
+        # Now obs_full has dimensions of len(date_range) by len(depths)
+        # Values are NaN wherever/whenever there are no observations
+
+        # There are some NaN values in the clarity and ice flag data.
+        # There are multiple ways to deal with NaN values in inputs:
+        # interpolate to fill them, skip over them, or remove sequences
+        # containing NaNs entirely. At this point, allow NaN values into the
+        # sites. Later code will handle NaN values.
```

Comment:
This comment block is a nice addition. I really like comments like this that orient the user to choices being made.